docs/CiderAllocator-use-guide.rst

```diff
@@ -0,0 +1,47 @@
+CiderAllocator use guide
+==========================================================================================================
+
+1 introduction to CiderAllocator
+We provide the allocation interface in CiderAllocator.h and two default implementations
+1.1
+CiderDefaultAllocator provide the default allocate and deallocate function via system library
+1.2
+AlignAllocator provide the function to perform memory alignment
+
+=========================================================================================================
+2 how to use
+user can construct an allocator from upstream memory pool, and then memory allocated will be in the pool.
+or use the CiderDefaultAllocator, AlignAllocator if they don't want to do it, 
+but this allocator must be present and passed throughout the calling logic
+because Only in this way can you know the memory usage during the entire operation for memory managerment
+
+---------------------------------------------------------------------------------------------------------
+A typical usage in CiderRunTimeModule
+2.1  add the member variables in .h file
+    `std::shared_ptr<CiderAllocator> allocator`
+    and add it in the constructor
+    `CiderRuntimeModule(
+      std::shared_ptr<CiderCompilationResult> ciderCompilationResult,
+      CiderCompilationOption ciderCompilationOption = CiderCompilationOption::defaults(),
+      CiderExecutionOption ciderExecutionOption = CiderExecutionOption::defaults(),
+      std::shared_ptr<CiderAllocator> allocator = std::make_shared<CiderDefaultAllocator>());`
+
+2.2 in .cpp implementation class
```

Comment:
seems each `CiderRuntimeModule` will have a self-managed allocator, which means we don't have a tree or child allocator yet. If so, I suggest to add some todo or limitation description in doc.

Reply:
> seems each `CiderRuntimeModule` will have a self-managed allocator, which means we don't have a tree or child allocator yet. If so, I suggest to add some todo or limitation description in doc.

Roger that，will add it later.